test/goth_test.exs

```diff
@@ -187,4 +257,36 @@ defmodule GothTest do
     {:ok, private_key}
     :public_key.pem_encode([:public_key.pem_entry_encode(:RSAPrivateKey, private_key)])
   end
+
+  defp http_client(options) do
+    validate_options(options)
+
+    {:ok,
+     %{
+       status: 200,
+       body: ~s|{"access_token":"dummy","expires_in":3599,"token_type":"Bearer"}|
+     }}
+  end
+
+  defp http_client_with_options(options) do
+    validate_options(options)
+
+    assert Keyword.has_key?(options, :test)
+    assert options[:test] === :"test custom http client with options"
+
+    {:ok,
+     %{
+       status: 200,
+       body: ~s|{"access_token":"dummy","expires_in":3599,"token_type":"Bearer"}|
+     }}
+  end
+
+  defp validate_options(options) do
+    assert Keyword.keyword?(options)
+
```

Comment:
this is not needed since if we don't have a keyword, any of the assertions below would fail too.

```suggestion
```